internal/bminventory/inventory.go

```diff
@@ -2289,6 +2291,160 @@ func (b *bareMetalInventory) DownloadHostLogs(ctx context.Context, params instal
+}
+
+func (b *bareMetalInventory) ListClusterManifests(ctx context.Context, params installer.ListClusterManifestsParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Listing manifests in cluster %s", params.ClusterID)
+
+	var cluster common.Cluster
+	if err := b.db.First(&cluster, identity.AddUserFilter(ctx, "id = ?"), params.ClusterID).Error; err != nil {
+		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	manifests, err := b.listManifests(ctx, cluster.ID, "manifests")
+	if err != nil {
+		return installer.NewListClusterManifestsInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	files, err := b.listManifests(ctx, cluster.ID, "openshift")
+	if err != nil {
+		return installer.NewListClusterManifestsInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	manifests = append(manifests, files...)
+	return installer.NewListClusterManifestsOK().WithPayload(manifests)
+}
+
+func (b *bareMetalInventory) listManifests(ctx context.Context, clusterID *strfmt.UUID, prefix string) (models.ListManifests, error) {
+	id := string(*clusterID)
+	objectName := filepath.Join(id, prefix)
+	files, err := b.objectHandler.ListObjects(ctx, &objectName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s from s3", objectName)
+	}
+	manifests := models.ListManifests{}
+	filePrefix := id + string(os.PathSeparator)
+	for _, file := range files {
+		manifests = append(manifests, &models.Manifest{FileName: strings.TrimPrefix(file, filePrefix)})
+	}
+	return manifests, nil
+}
+
+func (b *bareMetalInventory) DeleteClusterManifest(ctx context.Context, params installer.DeleteClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	fileName := filepath.Join(*params.Folder, params.FileName)
+	log.Infof("Deleting manifest %s from cluster %s", fileName, params.ClusterID)
+
+	var cluster common.Cluster
+	if err := b.db.First(&cluster, identity.AddUserFilter(ctx, "id = ?"), params.ClusterID).Error; err != nil {
+		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	objectName := filepath.Join(string(*cluster.ID), fileName)
+	exists, err := b.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to delete cluster manifest %s from s3", objectName)
+		return installer.NewDeleteClusterManifestInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	if !exists {
+		msg := fmt.Sprintf("Cluster manifest %s doesn't exists in cluster %s", fileName, cluster.ID)
+		log.Warn(msg)
+		return installer.NewDeleteClusterManifestNotFound().WithPayload(common.GenerateError(http.StatusNotFound, errors.New(msg)))
+	}
```

Comment:
I prefer that delete doesn't return error if the resource doesn't exist (idempotent).  The objectHandler method already takes care of this.

Reply:
fixed